src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -144,7 +146,7 @@ public FormValidation doCheckUrl(@AncestorInPath Item project,
                 return FormValidation.ok();
 
             // get git executable on master
-            final EnvVars environment = new EnvVars(System.getenv()); // GitUtils.getPollEnvironment(project, null, launcher, TaskListener.NULL, false);
+            final EnvVars environment = project.getEnvironment(Jenkins.getActiveInstance(), TaskListener.NULL);
```

Comment:
Ah.  I think I see why.  Unfortunately I consider this section a :bug: because we're getting the git tool configuration for a remote system but trying to validate connectivity on master.  (lines 151+). 
We're also potentially involving a remote system in the process (more failure potential). 

Given we're ignoring variables and treating those cases as a success, I think it's okay to just use System.getEnv here since it's only a UI-side check.  More complex problems will appear at polling anyway.